extensions/controllers/sandboxclaim_controller.go

```diff
@@ -0,0 +1,268 @@
+	if err := r.updateStatus(ctx, originalClaimStatus, claim); err != nil {
+		combinedErrors = errors.Join(combinedErrors, err)
+	}
+
+	return ctrl.Result{}, combinedErrors
+}
+
+func (r *SandboxClaimReconciler) updateStatus(ctx context.Context, oldStatus *extensionsv1alpha1.SandboxClaimStatus, claim *extensionsv1alpha1.SandboxClaim) error {
+	log := log.FromContext(ctx)
+
+	if reflect.DeepEqual(oldStatus, &claim.Status) {
+		return nil
+	}
+
+	if err := r.Status().Update(ctx, claim); err != nil {
+		log.Error(err, "Failed to update sandboxclaim status")
+		return err
+	}
+
+	return nil
+}
+
+func (r *SandboxClaimReconciler) computeReadyCondition(claim *extensionsv1alpha1.SandboxClaim, sandbox *v1alpha1.Sandbox, template *extensionsv1alpha1.SandboxTemplate, err error) metav1.Condition {
+	readyCondition := metav1.Condition{
+		Type:               string(sandboxv1alpha1.SandboxConditionReady),
+		ObservedGeneration: claim.Generation,
+		Status:             metav1.ConditionFalse,
+		Reason:             "MissingSandbox",
+		Message:            "Sandbox not found",
+	}
+
+	// Reconciler errors take precedence. They are expected to be transient.
+	if err != nil {
+		readyCondition.Reason = "ReconcilerError"
+		readyCondition.Message = "Error seen: " + err.Error()
+		return readyCondition
+	}
+
+	if sandbox == nil && template == nil {
+		readyCondition.Reason = "TemplateNotFound"
+		readyCondition.Message = fmt.Sprintf("SandboxTemplate %q not found", claim.Spec.TemplateRef.Name)
+		return readyCondition
+	}
+
+	if sandbox != nil {
+		sandboxReady := false
+		for _, condition := range sandbox.Status.Conditions {
+			if condition.Type == string(sandboxv1alpha1.SandboxConditionReady) {
+				if condition.Status == metav1.ConditionTrue {
+					sandboxReady = true
+				}
+				break
+			}
+		}
+		if sandboxReady {
+			readyCondition.Status = metav1.ConditionTrue
+			readyCondition.Reason = "SandboxReady"
+			readyCondition.Message = "Sandbox is ready"
+		} else {
+			readyCondition.Reason = "SandboxNotReady"
+			readyCondition.Message = "Sandbox is not ready"
+		}
+	}
+
+	return readyCondition
+}
+
+func (r *SandboxClaimReconciler) computeAndSetStatus(claim *extensionsv1alpha1.SandboxClaim, sandbox *v1alpha1.Sandbox, template *extensionsv1alpha1.SandboxTemplate, err error) {
+	// compute and set overall Ready condition
+	readyCondition := r.computeReadyCondition(claim, sandbox, template, err)
+	meta.SetStatusCondition(&claim.Status.Conditions, readyCondition)
+
+	if sandbox != nil {
+		claim.Status.SandboxStatus.Name = sandbox.Name
+		claim.Status.SandboxStatus.ServiceFQDN = sandbox.Status.ServiceFQDN
+		claim.Status.SandboxStatus.Service = sandbox.Status.Service
+	}
+}
+
+func (r *SandboxClaimReconciler) isOwnedByClaim(sandbox *v1alpha1.Sandbox, claim *extensionsv1alpha1.SandboxClaim) bool {
+	// Check if the existing sandbox is owned by this claim
+	owned := false
+	for _, ownerRef := range sandbox.OwnerReferences {
```

Comment:
There could be multiple owners, but only one controller. We should use controllerRef, i.e. set the controller flag to true. And then check the controllerRef here.